prow/config/tide.go

```diff
@@ -485,25 +486,42 @@ func parseTideContextPolicyOptions(org, repo, branch string, options TideContext
 	return option
 }
 
+// branchProtectionGitHubClient is the subset of the GitHub client
+// needed to access BranchProtection config
+type branchProtectionGitHubClient interface {
+	GetRef(org, repo, ref string) (string, error)
+}
+
 // GetTideContextPolicy parses the prow config to find context merge options.
 // If none are set, it will use the prow jobs configured and use the default github combined status.
 // Otherwise if set it will use the branch protection setting, or the listed jobs.
-func (c Config) GetTideContextPolicy(org, repo, branch string) (*TideContextPolicy, error) {
+func (c Config) GetTideContextPolicy(gitHubClient branchProtectionGitHubClient, gitClient *git.Client, org, repo, branch, headSHA string) (*TideContextPolicy, error) {
 	options := parseTideContextPolicyOptions(org, repo, branch, c.Tide.ContextOptions)
 	// Adding required and optional contexts from options
 	required := sets.NewString(options.RequiredContexts...)
 	requiredIfPresent := sets.NewString(options.RequiredIfPresentContexts...)
 	optional := sets.NewString(options.OptionalContexts...)
 
+	baseSHAGetter := func() (string, error) {
+		return gitHubClient.GetRef(org, repo, "heads/"+branch)
```

Comment:
Actually it looks like the status controller doesn't have the baseSHA. I'd be happy with the status controller looking up the baseSHA separately (although only once per pool per sync loop) since we won't have looked up the baseSHA for any repos where no PRs match the tide query and because the status controller is considered best effort.
We probably still want the sync controller to use a consistent baseSHA though?

Reply:
Yeah, the consistency is a good point here. I have updated the code to pass it on from the sync controller/use a closure in the status controller